api/src/main/java/com/epam/pipeline/manager/notification/NotificationManager.java

```diff
@@ -745,17 +745,18 @@ private NotificationMessage toMessage(final NotificationMessageVO messageVO) {
         final List<Long> copyUserIds = ListUtils.emptyIfNull(messageVO.getCopyUsers())
                 .stream()
                 .filter(StringUtils::isNotBlank)
-                .map(this::getUserByName)
+                .map(this::getUserByNameOrEmail)
                 .map(PipelineUser::getId)
                 .collect(Collectors.toList());
         message.setCopyUserIds(copyUserIds);
-        message.setToUserId(getUserByName(messageVO.getToUser()).getId());
+        message.setToUserId(getUserByNameOrEmail(messageVO.getToUser()).getId());
         return message;
     }
 
-    private PipelineUser getUserByName(final String username) {
-        final PipelineUser user = userManager.loadUserByName(username);
-        Assert.notNull(user, messageHelper.getMessage(MessageConstants.ERROR_USER_NAME_NOT_FOUND, username));
+    private PipelineUser getUserByNameOrEmail(final String usernameOrEmail) {
+        final PipelineUser user = Optional.ofNullable(userManager.loadUserByName(usernameOrEmail))
+                .orElse(userManager.loadUserByEmail(usernameOrEmail));
```

Comment:
It's better to only call the second method if the first one didn't succeed.

```suggestion
                .orElseGet(() -> userManager.loadUserByEmail(usernameOrEmail));
```